test/t8_cmesh_generator/t8_cmesh_example_sets.hxx

```diff
@@ -0,0 +1,60 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2024 the developers
+
+t8code is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+t8code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with t8code; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#ifndef T8_GTEST_CMESH_COMM_CREATOR_HXX
+#define T8_GTEST_CMESH_COMM_CREATOR_HXX
+
+#include <vector>
+
+#include "test/t8_cmesh_generator/t8_cmesh_parametrized_examples/t8_cmesh_new_prism_cake_param.hxx"
+#include "test/t8_cmesh_generator/t8_cmesh_parametrized_examples/t8_cmesh_new_from_class_param.hxx"
+#include "test/t8_cmesh_generator/t8_cmesh_parametrized_examples/t8_cmesh_new_bigmesh_param.hxx"
+#include "test/t8_cmesh_generator/t8_gtest_cmesh_cartestian_product.hxx"
+#include "test/t8_cmesh_generator/t8_gtest_cmesh_sum_of_sets.hxx"
+
+T8_EXTERN_C_BEGIN ();
+
+/**
+ * lambda to pass to an INSTANTIATE_TEST_SUITE_P to print the current base_example
+ * 
+ */
+auto pretty_print_base_example = [] (const testing::TestParamInfo<base_example *> &info) {
+  std::string name;
+  info.param->param_to_string (name);
+  return name;
+};
+
+namespace cmesh_list
+{
+std::vector<example_parameter_combinator *> cart_prod_vec
```

Comment:
Why do I need to create a Cartesian product of these three cmesh examples? What Cartesian product are we producing here? Refinement levels vs element classes, for example?

Reply:
I guuess the name is misleading here. Should be parameter_combinations instead of combinator. This vector holds three sets of cmeshes together with the Cartesian product of their parameters. If you have a look at the files where the sets are constructed you will see the Cartesian product